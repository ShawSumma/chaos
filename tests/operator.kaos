print "Relational operators on boolean variables"
bool a = true
bool b = false
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on boolean literals"
bool c = true == false
print c
del c
bool c = true != false
print c
del c
bool c = true > false
print c
del c
bool c = true < false
print c
del c
bool c = true >= false
print c
del c
bool c = true <= false
print c
del c

print "Relational operators on boolean variables vs. boolean literals"
bool a = true
bool c = a == false
print c
del c
bool c = a != false
print c
del c
bool c = a > false
print c
del c
bool c = a < false
print c
del c
bool c = a >= false
print c
del c
bool c = a <= false
print c
del c
del a

print "Relational operators on boolean literals vs. boolean variables"
bool b = false
bool c = true == b
print c
del c
bool c = true != b
print c
del c
bool c = true > b
print c
del c
bool c = true < b
print c
del c
bool c = true >= b
print c
del c
bool c = true <= b
print c
del c
del b

print "Relational operators on integer variables"
num a = 3
num b = 5
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on integer literals"
bool c = 3 == 5
print c
del c
bool c = 3 != 5
print c
del c
bool c = 3 > 5
print c
del c
bool c = 3 < 5
print c
del c
bool c = 3 >= 5
print c
del c
bool c = 3 <= 5
print c
del c

print "Relational operators on integer variables vs. integer literals"
num a = 3
bool c = a == 5
print c
del c
bool c = a != 5
print c
del c
bool c = a > 5
print c
del c
bool c = a < 5
print c
del c
bool c = a >= 5
print c
del c
bool c = a <= 5
print c
del c
del a

print "Relational operators on integer literals vs. integer variables"
num b = 5
bool c = 3 == b
print c
del c
bool c = 3 != b
print c
del c
bool c = 3 > b
print c
del c
bool c = 3 < b
print c
del c
bool c = 3 >= b
print c
del c
bool c = 3 <= b
print c
del c
del b

print "Relational operators on float variables"
num a = 13.5
num b = 763.78
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on float literals"
bool c = 13.5 == 763.78
print c
del c
bool c = 13.5 != 763.78
print c
del c
bool c = 13.5 > 763.78
print c
del c
bool c = 13.5 < 763.78
print c
del c
bool c = 13.5 >= 763.78
print c
del c
bool c = 13.5 <= 763.78
print c
del c

print "Relational operators on float variables vs. float literals"
num a = 13.5
bool c = a == 763.78
print c
del c
bool c = a != 763.78
print c
del c
bool c = a > 763.78
print c
del c
bool c = a < 763.78
print c
del c
bool c = a >= 763.78
print c
del c
bool c = a <= 763.78
print c
del c
del a

print "Relational operators on float literals vs. float variables"
num b = 763.78
bool c = 13.5 == b
print c
del c
bool c = 13.5 != b
print c
del c
bool c = 13.5 > b
print c
del c
bool c = 13.5 < b
print c
del c
bool c = 13.5 >= b
print c
del c
bool c = 13.5 <= b
print c
del c
del b

print "Relational operators on a mix of integer and boolean variables"
bool a = true
num b = 37
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on a mix of integer and boolean literals"
bool c = true == 37
print c
del c
bool c = true != 37
print c
del c
bool c = true > 37
print c
del c
bool c = true < 37
print c
del c
bool c = true >= 37
print c
del c
bool c = true <= 37
print c
del c

print "Relational operators on a mix of integer and float variables"
num a = 98
num b = 643.134
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on a mix of integer and float literals"
bool c = 98 == 643.134
print c
del c
bool c = 98 != 643.134
print c
del c
bool c = 98 > 643.134
print c
del c
bool c = 98 < 643.134
print c
del c
bool c = 98 >= 643.134
print c
del c
bool c = 98 <= 643.134
print c
del c

print "Relational operators on a mix of float and boolean variables"
bool a = false
num b = 3.14
bool c = a == b
print c
del c
bool c = a != b
print c
del c
bool c = a > b
print c
del c
bool c = a < b
print c
del c
bool c = a >= b
print c
del c
bool c = a <= b
print c
del c
del a
del b

print "Relational operators on a mix of float and boolean literals"
bool c = false == 3.14
print c
del c
bool c = false != 3.14
print c
del c
bool c = false > 3.14
print c
del c
bool c = false < 3.14
print c
del c
bool c = false >= 3.14
print c
del c
bool c = false <= 3.14
print c
del c

print "Relational operators on a mix of boolean variables and boolean/integer/float values"
bool a = true
bool b = false
bool c = a == 431
print c
del c
bool c = 3.14 != b
print c
del c
bool c = false > b
print c
del c
bool c = a < true
print c
del c
bool c = 1.0 >= 15
print c
del c
bool c = 14114 <= 345435.3414
print c
del c
del a
del b

print "Relational operators on a mix of boolean literals and boolean/integer/float values"
bool c = true == 431
print c
del c
bool c = 3.14 != false
print c
del c
bool c = false > false
print c
del c
bool c = true < true
print c
del c
bool c = 1.0 >= 15
print c
del c
bool c = 14114 <= 345435.3414
print c
del c


print "Logical operators on boolean variables"
bool a = true
bool b = false
bool c = a && b
print c
del c
bool c = a and b
print c
del c
bool c = a || b
print c
del c
bool c = a or b
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a
del b

print "Logical operators on boolean literals"
bool c = true && false
print c
del c
bool c = true and false
print c
del c
bool c = true || false
print c
del c
bool c = true or false
print c
del c
bool c = ! true
print c
del c
bool c = not true
print c
del c

print "Logical operators on boolean variables vs. boolean literals"
bool a = true
bool c = a && false
print c
del c
bool c = a and false
print c
del c
bool c = a || false
print c
del c
bool c = a or false
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a

print "Logical operators on boolean literals vs. boolean variables"
bool b = false
bool c = true && b
print c
del c
bool c = true and b
print c
del c
bool c = true || b
print c
del c
bool c = true or b
print c
del c
bool c = ! true
print c
del c
bool c = not true
print c
del c
del b

print "Logical operators on integer variables"
num a = 3
num b = 5
bool c = a && b
print c
del c
bool c = a and b
print c
del c
bool c = a || b
print c
del c
bool c = a or b
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a
del b

print "Logical operators on integer literals"
bool c = 3 && 5
print c
del c
bool c = 3 and 5
print c
del c
bool c = 3 || 5
print c
del c
bool c = 3 or 5
print c
del c
bool c = ! 3
print c
del c
bool c = not 3
print c
del c

print "Logical operators on integer variables vs. integer literals"
num a = 3
bool c = a && 5
print c
del c
bool c = a and 5
print c
del c
bool c = a || 5
print c
del c
bool c = a or 5
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a

print "Logical operators on integer literals vs. integer variables"
num b = 5
bool c = 3 && b
print c
del c
bool c = 3 and b
print c
del c
bool c = 3 || b
print c
del c
bool c = 3 or b
print c
del c
bool c = ! 3
print c
del c
bool c = not 3
print c
del c
del b

print "Logical operators on float variables"
num a = 13.5
num b = 763.78
bool c = a && b
print c
del c
bool c = a and b
print c
del c
bool c = a || b
print c
del c
bool c = a or b
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a
del b

print "Logical operators on float literals"
bool c = 13.5 && 763.78
print c
del c
bool c = 13.5 and 763.78
print c
del c
bool c = 13.5 || 763.78
print c
del c
bool c = 13.5 or 763.78
print c
del c
bool c = ! 13.5
print c
del c
bool c = not 13.5
print c
del c

print "Logical operators on float variables vs. float literals"
num a = 13.5
bool c = a && 763.78
print c
del c
bool c = a and 763.78
print c
del c
bool c = a || 763.78
print c
del c
bool c = a or 763.78
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a

print "Logical operators on float literals vs. float variables"
num b = 763.78
bool c = 13.5 && b
print c
del c
bool c = 13.5 and b
print c
del c
bool c = 13.5 || b
print c
del c
bool c = 13.5 or b
print c
del c
bool c = ! 13.5
print c
del c
bool c = not 13.5
print c
del c
del b

print "Logical operators on a mix of integer and boolean variables"
bool a = true
num b = 37
bool c = a && b
print c
del c
bool c = a and b
print c
del c
bool c = a || b
print c
del c
bool c = a or b
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a
del b

print "Logical operators on a mix of integer and boolean literals"
bool c = true && 37
print c
del c
bool c = true and 37
print c
del c
bool c = true || 37
print c
del c
bool c = true or 37
print c
del c
bool c = ! true
print c
del c
bool c = not true
print c
del c

print "Logical operators on a mix of integer and float variables"
num a = 98
num b = 643.134
bool c = a && b
print c
del c
bool c = a and b
print c
del c
bool c = a || b
print c
del c
bool c = a or b
print c
del c
bool c = ! a
print c
del c
bool c = not a
print c
del c
del a
del b

print "Logical operators on a mix of integer and float literals"
bool c = 98 && 643.134
print c
del c
bool c = 98 and 643.134
print c
del c
bool c = 98 || 643.134
print c
del c
bool c = 98 or 643.134
print c
del c
bool c = ! 98
print c
del c
bool c = not 98
print c
del c

print "Relational operators on a mix of boolean variables and boolean/integer/float values"
bool a = true
bool b = false
bool c = a && 431
print c
del c
bool c = 3.14 and b
print c
del c
bool c = false || b
print c
del c
bool c = a or true
print c
del c
bool c = ! 15
print c
del c
bool c = not 345435.3414
print c
del c
del a
del b

print "Relational operators on a mix of boolean literals and boolean/integer/float values"
bool c = true && 431
print c
del c
bool c = 3.14 and false
print c
del c
bool c = false || false
print c
del c
bool c = true or true
print c
del c
bool c = ! 15
print c
del c
bool c = not 345435.3414
print c
del c


print "Bitwise operators on integer variables"
num a = 5
num b = 9
num c = a & b
print c
del c
num c = a | b
print c
del c
num c = a ^ b
print c
del c
num c = ~ a
print c
del c
num c = b << 1
print c
del c
num c = b >> 1
print c
del c
del a
del b

print "Bitwise operators on integer literals"
num c = 5 & 9
print c
del c
num c = 5 | 9
print c
del c
num c = 5 ^ 9
print c
del c
num c = ~ 5
print c
del c
num c = 9 << 1
print c
del c
num c = 9 >> 1
print c
del c

print "Unary operators"
num a = 3
num b = 5
num c = (a + b) * a
print c
del a
del b
del c

num a = (3 + 5) * 3
print a
del a

num a = 3
num b = ++a
num c = ++13
print a
print b
print c
del b
del c

num b = a++
num c = 13++
print a
print b
print c
del b
del c

num b = --a
num c = --13
print a
print b
print c
del b
del c

num b = a--
num c = 13--
print a
print b
print c
del a
del b
del c


print "Long mathematical expressions"
print 3 + 5
print 145 + 24.8
print 0.32 + 0.42
print (3 + 5) * 2
print 6 * 4 / 12 + 72 / 8 - 9
print 3 + 6 * (11 + 1 - 4) / 8 * 2
print 3 + (6 * (11 + 1 - 4)) / 8 * 2
print 8 / 2 * (2 + 2)
print (17 - 6 / 2) + 4 * 3

print "Operator precedence for variables"
num n = 3
num a = 2 * n - 2
print a
del n
del a
