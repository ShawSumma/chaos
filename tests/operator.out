false
true
true
false
true
false
true
false
false
false
true
true
true
false
false
false
true
true
true
false
false
false
true
true
true
false
false
false
true
true
false
true
false
true
false
true
false
true
false
false
false
true
false
false
true
true
false
false
true
true
true
true
false
false
true
true
true
true
false
false
true
true
true
true
false
false
true
true
true
true
false
false
true
true
true
true
false
false
true
false
false
true
false
false
1
13
12
-6
18
4
11
24
4
4
4
3
